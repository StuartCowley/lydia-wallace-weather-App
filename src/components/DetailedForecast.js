import "../styles/DetailedForecast.css";

import React from "react";
import WeatherIcon from "react-icons-weather";

import PropTypes from "prop-types";

function DetailedForecast({ forecasts, selectedDate }) {
  const formattedDate = new Date(selectedDate).toDateString();
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  return (
    <div className="detailed-forecast">
      <b>
        <div className="detailedForecast__date">{`Details for ${formattedDate}`}</div>
      </b>
      <div className="forecastSummary__icon" data-testid="forecast-icon">
        <WeatherIcon
          name="owm"
          iconId={`${selectedForecast.icon}`}
          flip="horizontal"
          rotate="90"
        />{" "}
        {selectedForecast.description}
      </div>
      <div className="detailedForecast__maxTemperature">{`Max: ${selectedForecast.temperature.max}°C`}</div>
      <div className="detailedForecast__minTemperature">{`Min: ${selectedForecast.temperature.min}°C`}</div>
      <div className="detailedForecast__humidity">{`Humidity: ${selectedForecast.humidity}%`}</div>
      <div className="detailedForecast__windSpeed">{`Wind Speed: ${selectedForecast.wind.speed}mph`}</div>
      <div className="detailedForecast__windDirection">{`Wind Direction: ${selectedForecast.wind.direction}`}</div>
    </div>
  );
}

DetailedForecast.propTypes = {
  selectedDate: PropTypes.number.isRequired,
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
      wind: PropTypes.shape({
        speed: PropTypes.number,
        direction: PropTypes.string,
      }),
      humidity: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.number,
    })
  ).isRequired,
};

export default DetailedForecast;
