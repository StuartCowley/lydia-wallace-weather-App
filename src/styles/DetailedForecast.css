.detailed-forecast {
    padding: 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
}