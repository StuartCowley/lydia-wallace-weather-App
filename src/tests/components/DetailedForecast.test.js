import React from "react";
import { render } from "@testing-library/react";
import DetailedForecast from "../../components/DetailedForecast";

describe("Detailed Forecast", () => {
  const validProps = [
    {
      date: 1525046400000,
      temperature: {
        max: 15,
        min: 10,
      },
      wind: {
        speed: 11,
        direction: "se",
      },
      humidity: 67,
      description: "test",
      icon: 800,
    },
    {
      date: 1525046400022,
      temperature: {
        max: 10,
        min: 5,
      },
      wind: {
        speed: 19,
        direction: "n",
      },
      humidity: 78,
      description: "test2",
      icon: 800,
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(
      <DetailedForecast
        forecasts={validProps}
        selectedDate={validProps[0].date}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // further test plans:
  // it("renders correct values for props", () => {});
  // it("changes based on which date is selected", () => {});
});
