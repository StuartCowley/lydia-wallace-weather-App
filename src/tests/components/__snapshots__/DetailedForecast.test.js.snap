// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detailed Forecast renders correctly 1`] = `
<DocumentFragment>
  <div
    class="detailed-forecast"
  >
    <b>
      <div
        class="detailedForecast__date"
      >
        Details for Mon Apr 30 2018
      </div>
    </b>
    <div
      class="detailedForecast__maxTemperature"
    >
      Max Temperature: 15°C
    </div>
    <div
      class="detailedForecast__minTemperature"
    >
      Min Temperature: 10°C
    </div>
    <div
      class="detailedForecast__humidity"
    >
      Humidity: 67%
    </div>
    <div
      class="detailedForecast__windSpeed"
    >
      Wind Speed: 11mph
    </div>
    <div
      class="detailedForecast__windDirection"
    >
      Wind Direction: se
    </div>
  </div>
</DocumentFragment>
`;
